src/main/java/com/example/petbuddybackend/controller/ChatWebSocketController.java

```diff
@@ -0,0 +1,50 @@
+package com.example.petbuddybackend.controller;
+
+import com.example.petbuddybackend.dto.chat.ChatMessageDTO;
+import com.example.petbuddybackend.dto.chat.ChatMessageSent;
+import com.example.petbuddybackend.entity.user.Role;
+import com.example.petbuddybackend.service.chat.ChatService;
+import com.example.petbuddybackend.utils.header.HeaderUtils;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.messaging.handler.annotation.*;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+
+import java.security.Principal;
+import java.time.ZoneId;
+import java.util.Map;
+import java.util.Optional;
+
+@Controller
+@RequiredArgsConstructor
+public class ChatWebSocketController {
+
+    @Value("${header-name.timezone}")
+    private String TIMEZONE_HEADER_NAME;
+
+    @Value("${header-name.role}")
+    private String ROLE_HEADER_NAME;
+
+    private final ChatService chatService;
+
+    @PreAuthorize("isAuthenticated()")
+    @MessageMapping("/chat/{chatId}")
+    @SendTo("/topic/messages/{chatId}")
+    public ChatMessageDTO sendChatMessage(
+            @DestinationVariable Long chatId,
+            @Valid @Payload ChatMessageSent message,
+            @Headers Map<String, Object> headers,
+            Principal principal
+    ) {
+        String username = principal.getName();
+        Role acceptRole = HeaderUtils.getHeaderSingleValue(headers, ROLE_HEADER_NAME, Role.class);
+        Optional<String> acceptTimeZone = HeaderUtils.getOptionalHeaderSingleValue(headers, TIMEZONE_HEADER_NAME, String.class);
+
+        // Full support for time zones will added in next PR
+        return acceptTimeZone.isPresent() ?
+                chatService.createMessage(chatId, username, message, acceptRole, ZoneId.of("Europe/Warsaw")) :
```

Comment:
Europe/Warsaw do application.yml raczej

Reply:
Wiesz co zostawiłem bo mam problem z tymi time zoneami i ten hardcode usunę jak to doimplementuje